beetsplug/bandcamp/__init__.py

```diff
@@ -218,9 +218,11 @@ def candidates(
 
         if album == "":
             search = {
-                "query": artist + " - " + item.title,
-                "artist": artist,
-                "label": label,
+                "query": item.artist + " - " + item.title
+                if item.artist != ""
+                else item.title,
```

Comment:
You can use this sort of a thing here
```suggestion
                "query": " - ".join(filter(None, [artist, title])),
```

This removes the need for a conditional which is nice

Reply:
Actually, it would be good to isolate empty album handling logic into a separate conditional since I get a bit confused looking at it in the current state.

```python
        if album:
            query = album
            search_type = "a"
        else:
            query = " - ".join(filter(None, [artist, item.title]))
            search_type = ""

        search = {
            "query": query,
            "artist": artist,
            "label": label,
            "search_type": search_type,
        }
```